### ISSUE TITLE
Fix FP S3353 (`prefer-const`): not respecting ES6 object destructuring assignment

### ISSUE THREAD

**Issue by manovotny** [SonarSource/SonarJS#1936]
**I want to report a bug.**

**SonarJS version:** 5.2.1.7778
**SonarQube version:** 8
**SonarLint version:** N/A

**Rule key:** 

[S3353 (aka. UnchangedLetVariableCheck)](https://github.com/SonarSource/SonarJS/blob/1916267988093cb5eb1d0b3d74bb5db5c0dbedec/javascript-checks/src/main/java/org/sonar/javascript/checks/UnchangedLetVariableCheck.java)

**Reproducer** 

```
let a = 0;

if (someCondition) {
    {value: a} = await someCall();
}

console.log(a);
```

SonarJS will state that `a`'s declaration needs to be a `const`, even though it's reset in the `if` condition.

**Expected behavior**

It seems that SonarJS doesn't understand [ES6 object destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assigning_to_new_variable_names). 

It should be noted that eslint's [prefer-const](https://eslint.org/docs/rules/prefer-const) does not flag this as an error.

Expected behavior is that S3353 (aka. UnchangedLetVariableCheck) would not see this as an error.

**Comment by saberduck** [comment]
hello @manovotny ,

can you try with SonarJS 6.2? You should be able to upgrade SonarJS analyzer via marketplace.

**Comment by ilia-kebets-sonarsource** [comment]
The code above, [once fixed](https://flaviocopes.com/javascript-destructure-object-to-existing-variable/), does not raise an issue:

```js
let a = 0;

if (some condition) {
    ({value: a} = await someCall());
}

console.log(a);
```
